### PR TITLE
Some alterations for the MAX32630FTHR

### PR DESCRIPTION
**Pull request by idea--list**
Added some pin definitions for the MAX32630FTHR, changed naming of the analog pins to be on par with the pinout diagram of this board.

Added a type definition in Wire.cpp for MAX32630FTHR as the compiler refuses to compile without that.

The other 2 targets might also need these changes, however i do not have those at my hands , so i can not test those.

### CODE REVIEW

File: arm/variants/max3263x_fthr/variant.h [hunk 1]
```diff
@@ -79,16 +80,25 @@
...
+// Standardized button names
+#define BUTTON1 SW1
+
+
 // Analog pins
 // -----------
-#define PIN_A0  49
-#define PIN_A1  50
-#define PIN_A2  51
-#define PIN_A3  52
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h:87]: These pin names must be kept to maintain compatibility with existing programs. Please add them back.
As per Arduino naming convention, A0 through A4 should be used.

File: arm/variants/max3263x_fthr/variant.h [hunk 2]
```diff
@@ -115,7 +125,7 @@ static const uint8_t MOSI0 = PIN_SPI0_MOSI;
 static const uint8_t MISO0 = PIN_SPI0_MISO;
 static const uint8_t SCK0  = PIN_SPI0_SCK;
 
-// SPI1
+// SPI1: XIP Flash (Optional, board is produced without an XIP flash chip, so you need to buy&solder one)
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h]: Update comment:
// SPI1: Option to add XIP Flash on the board

File: arm/variants/max3263x_fthr/variant.h [hunk 3]
```diff
@@ -145,24 +155,38 @@ static const uint8_t SCK2  = PIN_SPI2_SCK;
 static const uint8_t SDA = PIN_WIRE_SDA;
 static const uint8_t SCL = PIN_WIRE_SCL;
 
-// Wire0
+// Wire0 (??)
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h]: Update comment:
// Wire0: Not connected (I2CM0)

File: arm/variants/max3263x_fthr/variant.h [hunk 4]
```diff
@@ -145,24 +155,38 @@ static const uint8_t SCK2  = PIN_SPI2_SCK;
...
 
-// Wire1: Feather pins
+// Wire1: Feather pins (I2CM1)
 #define PIN_WIRE1_SDA   28
 #define PIN_WIRE1_SCL   29
 static const uint8_t SDA1 = PIN_WIRE1_SDA;
 static const uint8_t SCL1 = PIN_WIRE1_SCL;
 
-// Wire2: Feather pins
+// Wire2: Feather pins (I2CM2)
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h]: Update comment:
// Wire2: Feather pins, BMI160, MAX14690 (I2CM2)

**Review by khariya — CHANGES_REQUESTED** [review]

File: arm/libraries/Wire/src/Wire.cpp [hunk 5]
```diff
@@ -116,7 +117,7 @@ void TwoWire::setClock(uint32_t clock)
     }
 
     // Update current_speed if clock is valid, or set to default speed
-    current_speed = speed;
+    i2cm_speed_t current_speed = speed;
```
Comment (khariya) [inline arm/libraries/Wire/src/Wire.cpp]: Why is this required?
I'm able to compile successfully on Arduino 1.8.12.

File: arm/variants/max3263x_fthr/variant.h [hunk 6]
```diff
@@ -64,6 +64,7 @@
 #define SERIAL_PORT_HARDWARE_OPEN2  Serial3
 
 #define PIN_NC      -1
+#define NOT_CONNECTED PIN_NC
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h:67]: Is this required?

File: arm/variants/max3263x_fthr/variant.h [hunk 7]
```diff
@@ -79,16 +80,25 @@
...
 #define LED_BUILTIN RED_LED
 
+// Push button
+// ----
+#define SW1 P2_3
+#define SW2 NOT_CONNECTED
+#define SW3 NOT_CONNECTED
+// Standardized button names
+#define BUTTON1 SW1
+
```
Comment (khariya) [inline arm/variants/max3263x_fthr/variant.h]: Are these required?

**Review by khariya — CHANGES_REQUESTED** [review]

File: arm/variants/max3263x_fthr/variant.h (hunk 1 above)
Comment (idea--list) [reply arm/variants/max3263x_fthr/variant.h:87]: readded these definitions.

File: arm/variants/max3263x_fthr/variant.h (hunk 2 above)
Comment (idea--list) [reply arm/variants/max3263x_fthr/variant.h]: changed to your comment as that is shorter

File: arm/variants/max3263x_fthr/variant.h (hunk 3 above)
Comment (idea--list) [reply arm/variants/max3263x_fthr/variant.h]: Took your comment. Originally i was unsure if this port was connected.

File: arm/variants/max3263x_fthr/variant.h (hunk 6 above)
Comment (idea--list) [reply arm/variants/max3263x_fthr/variant.h:67]: This is not absolutely required, however if someone uses the longer version in their mBed code, it is easier to port the code to Arduino if also the longer version is defined.

File: arm/libraries/Wire/src/Wire.cpp (hunk 5 above)
Comment (idea--list) [reply arm/libraries/Wire/src/Wire.cpp]: Ok our results seem to be inconsistent. If i do not specify the type of `current_speed` then i get this error in Arduino 1.8.12:
image
So at least on my system it seems to be absolutely required.
My system runs on Windows 10 Pro x64 build 18363 and have also installed Mbed Studio 0.8.1 and even MatLab 2019b and its Arduino plugin, so maybe those affect somehow the behaviour.

File: arm/variants/max3263x_fthr/variant.h (hunk 7 above)
Comment (idea--list) [reply arm/variants/max3263x_fthr/variant.h]: I think defining SW1 and BUTTON1 is useful as people may use those in their original mBed code as these names are shown on the pinout diagram of this board.
As for SW2 and SW3 i included them as they are there in the pindefinition for mBed of this board, but as these pins of the µCU are not lead out on the board, people will not use these names in their code, we can just drop them.

File: arm/variants/max3263x_fthr/variant.h (hunk 6 above)
Comment (khariya) [reply arm/variants/max3263x_fthr/variant.h:67]: Got it.

File: arm/libraries/Wire/src/Wire.cpp (hunk 5 above)
Comment (khariya) [reply arm/libraries/Wire/src/Wire.cpp]: On further investigation, the `current_speed` is already declared in a system library as a global variable. This change will re-declare it as a local variable.
I'm unable to reproduce the issue on my end so I don't know why you see that error.
Tried on Ubuntu 16.04 LTS and Windows 10 X64 Pro, Build 16299.

Please remove this change.

Comment (idea--list) [reply arm/libraries/Wire/src/Wire.cpp]: Reverted back to original.
